test.cpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <assert.h>
+#include "test.h"
+
+void TestColorCode::testNumberToPair(int pairNumber,
+	MajorColor expectedMajor,
+	MinorColor expectedMinor)
+{
+	ColorPair colorPair =
+		ColorCodeConverter::GetColorFromPairNumber(pairNumber);
+	std::cout << "Got pair " << colorPair.ColorPairToString() << std::endl;
+	assert(colorPair.getMajorColor() == expectedMajor);
+	assert(colorPair.getMinorColor() == expectedMinor);
+	return;
+}
+
+void TestColorCode::testPairToNumber(
+	MajorColor major,
+	MinorColor minor,
+	int expectedPairNumber)
+{
+	int pairNumber = ColorCodeConverter::GetPairNumberFromColor(major, minor);
+	std::cout << "Got pair number " << pairNumber << std::endl;
+	assert(pairNumber == expectedPairNumber);
+	return;
+}
+
+void TestColorCode::testPrintAllColorPairsWithNumber()
```

Comment:
`testPrintAllColorPairsWithNumber` is actually the printing functionality itself. It is part of production code, isn't it? A user who wants this functionality is forced to pick up test code as well...

Reply:
Yes I understood. Moved the functionality to the production part of the code.
Thanks for giving me the feedback. It gives some meaningful insights to me.